generated/nifgen/nifgen/session.py

```diff
@@ -3083,6 +3083,7 @@ def __init__(self, resource_name, channel_name=None, reset_device=False, options
             session (nifgen.Session): A session object representing the device.
 
         '''
+        self._grpc_options = _grpc_options
```

Comment:
This seems really odd to modify production code this way.

Reply:
If we don't want to enforce all implementations to have it, shouldn't we just use `hasattr(_close_on_exit)`?

Reply:
This `self._grpc_options` is needed for the code in `__exit__`, so that it only checks for `self._interpreter._close_on_exit` if `self._interpreter` is a GrpcStubInterpreter -- because only GrpcStubInterpreter has a `_close_on_exit` member. Some alternatives:
1. I could instead change `__exit__` to check `isinstance(self._interpreter, GrpcStubInterpreter)`, or 
2. I could change the `_close_on_exit` check to be `getattr(self._interpreter, '_close_on_exit', False)`, or
3. I could add `self._close_on_exit` to the LibraryInterpreter constructor. (Or similar, I could add a close_on_exit function to both Interpreters, and for LibraryInterpreter always return True.)

I'm open to any of those options, I just chose what I saw to be the least disruptive.

Reply:
My PR also added `_close_on_exit = True` to [the library interpreter](https://github.com/ni/nimi-python/blob/aa77b9334344dcb479af2a34ce155973192adca7/generated/nifgen/nifgen/_library_interpreter.py#L490). Why not propagate it elsewhere?

Reply:
Just saw your `hasattr` comment. That would be similar to option 2, and that would be fine, too.

Reply:
Ah, I see now that it is indeed on the library interpreter, too. The problem is just that it's only set after init is called. But if init fails, session tries to query it anyway. So that's a different sort of bug. Okay, I'll fix it.

Reply:
Ah, but here's why that shouldn't be an issue. The `__exit__` call comes only after `__enter__` which can only be done by putting the new session into a context manager after `__init__` has succeeded. If the object isn't initialized, it won't be added to the context manager to start with.

Reply:
Okay, makes sense. Done.